application/src/main/java/org/togetherjava/tjbot/commands/help/HelpSystemHelper.java

```diff
@@ -333,6 +333,55 @@ private void executeUncategorizedAdviceCheck(long threadChannelId, long authorId
         }).queue();
     }
 
+    void scheduleNoActivityAdviceCheck(long threadChannelId, long authorId) {
+        SERVICE.schedule(() -> {
+            try {
+                executeNoActivityAdviceCheck(threadChannelId, authorId);
+            } catch (Exception e) {
+                logger.warn(
+                        "Unknown error during a no activity advice check on thread {} by author {}.",
+                        threadChannelId, authorId, e);
+            }
+        }, SEND_NO_ACTIVITY_ADVICE_AFTER_MINUTES, TimeUnit.MINUTES);
+    }
+
+    private void executeNoActivityAdviceCheck(long threadChannelId, long authorId) {
+        logger.debug("Executing no activity advice check for thread {} by author {}.",
+                threadChannelId, authorId);
+
+        ThreadChannel threadChannel = jda.getThreadChannelById(threadChannelId);
+        if (threadChannel == null) {
+            logger.debug(
+                    "Channel for no activity advice check seems to be deleted (thread {} by author {}).",
+                    threadChannelId, authorId);
+            return;
+        }
+
+        if (threadChannel.isArchived()) {
+            logger.debug(
+                    "Channel for no activity advice check is archived already (thread {} by author {}).",
+                    threadChannelId, authorId);
+            return;
+        }
+        if (threadChannel.getHistory()
+            .getRetrievedHistory()
+            .stream()
+            .map(Message::getAuthor)
+            .map(User::getIdLong)
+            .anyMatch(id -> id == authorId)) {
```

Comment:
try to avoid having long ifs like this. its really hard to read. instead, move the thing into a helper method or at least into a well named boolean var

Reply:
Done